src/pages/collections/[collectionId]/index.tsx

```diff
@@ -0,0 +1,122 @@
+import { useEffect, useState } from 'react';
+
+import { GetStaticPaths, GetStaticProps } from 'next';
+import useTranslation from 'next-translate/useTranslation';
+import { useRouter } from 'next/router';
+import useSWRInfinite from 'swr/infinite';
+
+import layoutStyles from '../../index.module.scss';
+
+import styles from './index.module.scss';
+
+import { isLoggedIn } from '@/utils/auth/login';
+import CollectionDetail from 'src/components/Collection/CollectionDetail/CollectionDetail';
+import Button from 'src/components/dls/Button/Button';
+import DataContext from 'src/contexts/DataContext';
+import { privateFetcher } from 'src/utils/auth/api';
+import { makeGetBookmarkByCollectionId } from 'src/utils/auth/apiPaths';
+import { getAllChaptersData } from 'src/utils/chapter';
+import { GetBookmarkCollectionsIdResponse } from 'types/auth/GetBookmarksByCollectionId';
+
+const defaultSortOptionId = 'recentlyAdded';
+
+const CollectionDetailPage = ({ chaptersData }) => {
+  const { t } = useTranslation();
+  const router = useRouter();
+  const collectionId = router.query.collectionId as string;
+
+  useEffect(() => {
+    if (!isLoggedIn()) {
+      router.replace('/');
+    }
+  }, [router]);
+
+  const [sortBy, setSortBy] = useState(defaultSortOptionId);
+
+  const onSortByChange = (newSortByVal) => {
+    setSortBy(newSortByVal);
+  };
+
+  const getKey = (pageIndex, previousPageData) => {
```

Comment:
Would be awesome to add documentation to this method since it's a bit hard to follow.